@xen-orchestra/xapi/host.mjs

```diff
@@ -31,7 +33,21 @@ class Host {
    *
    * @param {string} ref - Opaque reference of the host
    */
-  async smartReboot($defer, ref) {
+  async smartReboot($defer, ref, bypassBlockedSuspend = false) {
+    const residentVmRefs = await this.getField('host', ref, 'resident_VMs')
+    const vmsWithSuspendBlocked = await asyncMap(residentVmRefs, ref => this.getRecord('VM', ref)).filter(
+      vm => vm.power_state !== 'Halted' && vm.power_state !== 'Suspend' && vm.blocked_operations.suspend !== undefined
+    )
+
+    if (!bypassBlockedSuspend && vmsWithSuspendBlocked.length > 0) {
+      throw operationBlocked({ code: 'suspend', objectId: vmsWithSuspendBlocked.map(vm => vm.uuid) })
+    }
+
+    await asyncEach(vmsWithSuspendBlocked, vm => {
+      $defer(() => this.editVm(vm, { blocked_operations: { suspend: vm.blocked_operations.suspend } }))
```

Comment:
You cannot use `editVm` here. It is defined in `xo-server` and not in this class (`@xen-orchestra/xapi`) which inherits from `xen-api`.

You can use `vm.update_blocked_operations('suspend', vm.blocked_operations.suspend ?? null)`. (`null` is necessary to remove the entry, `undefined` will not work).

Reply:
Because you are using it in the context of `xo-server/Xapi` which inherits from this class.

But `@xen-orchestra/xapi` is used in other contexts and must be self-contained.